lib/privs.c

```diff
@@ -496,26 +501,26 @@ struct zebra_privs_t *_zprivs_raise(struct zebra_privs_t *privs,
 	int save_errno = errno;
 	struct zebra_privs_refs_t *refs;
 
-	if (!privs)
-		return NULL;
-
 	/*
 	 * Serialize 'raise' operations; particularly important for
 	 * OSes where privs are process-wide.
 	 */
-	frr_with_mutex (&(privs->mutex)) {
-		/* Locate ref-counting object to use */
-		refs = get_privs_refs(privs);
+	pthread_mutex_lock(&zprivs_mutex);
+	if (!privs)
+		return NULL;
+	;
 
-		if (++(refs->refcount) == 1) {
-			errno = 0;
-			if (privs->change(ZPRIVS_RAISE)) {
-				zlog_err("%s: Failed to raise privileges (%s)",
-					 funcname, safe_strerror(errno));
-			}
-			errno = save_errno;
-			refs->raised_in_funcname = funcname;
+	/* Locate ref-counting object to use */
+	refs = get_privs_refs(privs);
+
+	if (++(refs->refcount) == 1) {
```

Comment:
Stupid question. Is it even possible for this to happen anymore with the mutex being global?

Also if `privs_per_process` is false, we don't need the global mutex?